wade-review.md

```diff
@@ -0,0 +1,596 @@
+
+
+```java
+public class GitHubPrRatio {
+    private static final Logger LOG = LoggerFactory.getLogger(GitHubPrRatio.class);
+
+}
+```
+A logger for verifying that the data’s coming in, 
+```java
+public class GitHubPrRatio {
+    // start with public class, then fill in variables in screen recording
+    private static final Logger LOG = LoggerFactory.getLogger(GitHubPrRatio.class);
+    static final Serde<GitHubPRStateCounter> prStateCounterSerde = StreamsSerde.serdeFor(GitHubPRStateCounter.class);
+    final Serializer<JsonNode> jsonSerializer = new JsonSerializer();
+    final Deserializer<JsonNode> jsonDeserializer = new JsonDeserializer();
+    final Serde<JsonNode> jsonNodeSerde = Serdes.serdeFrom(jsonSerializer, jsonDeserializer);
+}
+```
+Variables working towards creating my custom serde,
+```java
+public class GitHubPrRatio {
+    // start with public class, then fill in variables in screen recording
+    private static final Logger LOG = LoggerFactory.getLogger(GitHubPrRatio.class);
+    static final Serde<GitHubPRStateCounter> prStateCounterSerde = StreamsSerde.serdeFor(GitHubPRStateCounter.class);
+    final Serializer<JsonNode> jsonSerializer = new JsonSerializer();
+    final Deserializer<JsonNode> jsonDeserializer = new JsonDeserializer();
+    final Serde<JsonNode> jsonNodeSerde = Serdes.serdeFrom(jsonSerializer, jsonDeserializer);
+    static final String STORE_NAME = "pr-state";
+    static final String NO_STATE = "no-state";
+    static final String INPUT_TOPIC = "github-pull_requests";
+    static final String OUTPUT_TOPIC = "github-pull_requests"
+
+}
+```
+And some strings – my storename, a no state variable, and some input and output topic names. 
+
+## The main method
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+}
+```
+
+Let’s wrap all this up in our main method. 
+
+```java
+    public static Properties loadEnvProperties(String fileName) throws IOException {
+        final Properties allProps = new Properties();
+        try (final FileInputStream input = new FileInputStream(fileName)) {
+            allProps.load(input);
+        }
+        return allProps;
+    }
+```
+
+I’ll load the properties – and hold on, let’s make the method that loads them. This method points at the path to our properties file and returns the contents.
+
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-github-87");
+        props.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+}
+
+Let’s add a couple things to the properties down here– consumer group id, set the offset to earliest, and a max cache bytes config.
```

Comment:
Why aren't these in the properties file?

Reply:
much of the code in this project was an amalgamation of tutorials and course exercises. this piece inherits the idea of keeping these pieces of configuration here for teaching emphasis. but I don't intend to go into them much. on the fence about whether to move it.

Reply:
If you don't intend to go through it, and it can be moved to the props file, I'd do that. It will reduce the amount of stuff the viewer has to absorb while watching the video.